### PR TITLE
Implemented user account cancelling scenario on Rspec

### PR DESCRIPTION
**Pull request by pavel-temtsunik**

### CODE REVIEW

File: spec/cancelling_helper.rb [hunk 1]
```diff
@@ -0,0 +1,37 @@
+module SignUpHelper
```
Comment (SirNicholas) [inline spec/cancelling_helper.rb]: module name is SignUpHelper, but file name is cancelling_helper.rb . Why so?

File: spec/bvt/functionality/account/cancelling_spec.rb [hunk 2]
```diff
@@ -0,0 +1,15 @@
+require 'spec_helper'
+
+feature "Account cancelling" do
+  scenario "Visitor can cancel his own account if he is signed in" do
```
Comment (SirNicholas) [inline spec/bvt/functionality/account/cancelling_spec.rb]: Visitor - is a person, that doesn't have any credentials on visited site. In this case, It should be like this:
scenario "User can cancel his own account if he is signed in"

File: spec/cancelling_helper.rb [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+  def user_signs_up
```
Comment (SirNicholas) [inline spec/cancelling_helper.rb]: Method should have one parameter: user
So we can use this method in future tests to Sign Up any user we want.
In current state this method will always register some generated user

File: spec/cancelling_helper.rb [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+
+    ConfirmationInstructionEmail.
+        find_by_recipient(user.email).
+        confirm_my_account
+    expect(LoginPage.given.text).to include('Your account was successfully confirmed.')
+
+
+  end
+
+  def user_logs_in_into_system
```
Comment (SirNicholas) [inline spec/cancelling_helper.rb]: Method should have one parameter: user
So we can login with different users with just one method

File: spec/cancelling_helper.rb [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+  def user_signs_up
+    user = Gen::given_user_by_number(0)
+    SignUpPage.
+        open.fill_form(
+        user_name: user.full_name,
+        email: user.email,
+        password: user.password,
+        password_confirmation: user.password)
+        .submit_form
+    expect(HomePage).to_not be_authenticated
```
Comment (SirNicholas) [inline spec/cancelling_helper.rb]: This method is a helper method, so there is no need in additional checks. We should check only that user has been successfully registered

File: spec/cancelling_helper.rb (hunk 1 above)
Comment (pavel-temtsunik) [inline spec/cancelling_helper.rb]: +

File: spec/bvt/functionality/account/cancelling_spec.rb (hunk 2 above)
Comment (pavel-temtsunik) [inline spec/bvt/functionality/account/cancelling_spec.rb]: +

File: spec/sign_up_helper.rb [hunk 6]
```diff
@@ -0,0 +1,31 @@
...
+        email: user.email,
+        password: user.password,
+        password_confirmation: user.password)
+        .submit_form
+
+    ConfirmationInstructionEmail.
+        find_by_recipient(user.email).
+        confirm_my_account
+    expect(LoginPage.given.text).to include('Your account was successfully confirmed.')
+
```
Comment (SirNicholas) [inline spec/sign_up_helper.rb]: Удали пожалуста лишние переносы строк

File: spec/spec_helper.rb [hunk 7]
```diff
@@ -1,6 +1,7 @@
 require 'rspec'
 require 'capybara/rspec'
 require_relative '../boot'
+require_relative '../spec/sign_up_helper'
```
Comment (SirNicholas) [inline spec/spec_helper.rb]: Путь к файлу прописан не верно. Не нужно еще раз писать путь к папке spec, потому-что ты уже в ней находишься

File: spec/bvt/functionality/account/cancelling_spec.rb [hunk 8]
```diff
@@ -0,0 +1,16 @@
+require 'spec_helper'
+
+feature "Account cancelling" do
+  scenario "User can cancel his own account if he is signed in" do
+    user_signs_up (Gen::given_user_by_number(0))
```
Comment (SirNicholas) [inline spec/bvt/functionality/account/cancelling_spec.rb]: Не нужно писать пробел после имени метода и перед скобками с параметрами

File: spec/bvt/functionality/account/cancelling_spec.rb [hunk 9]
```diff
@@ -0,0 +1,16 @@
...
+    user_logs_in_into_system (Gen::given_user_by_number(0))
+
+    EditAccountPage.open
```
Comment (SirNicholas) [inline spec/bvt/functionality/account/cancelling_spec.rb]: Зачем переводить вызов метода cancel_my_account в другую строку?

File: spec/bvt/functionality/account/cancelling_spec.rb [hunk 10]
```diff
@@ -0,0 +1,16 @@
...
+                   .cancel_my_account
+    accept_alert
```
Comment (SirNicholas) [inline spec/bvt/functionality/account/cancelling_spec.rb]: Где этот метод описан?

File: spec/bvt/functionality/account/cancelling_spec.rb (hunk 9 above)
Comment (romikoops) [inline spec/bvt/functionality/account/cancelling_spec.rb]: If it is easier to read and support, not problem

File: spec/sign_up_helper.rb (hunk 6 above)
Comment (romikoops) [inline spec/sign_up_helper.rb]: the same not problem

File: spec/spec_helper.rb (hunk 7 above)
Comment (romikoops) [inline spec/spec_helper.rb]: Please replace with
Dir[Rails.root.join("spec/support/*/.rb")].each {|f| require f}

File: spec/spec_helper.rb [hunk 11]
```diff
@@ -1,6 +1,8 @@
 require 'rspec'
 require 'capybara/rspec'
 require_relative '../boot'
+Dir[File.join(File.dirname(__FILE__), 'support', '**', '*.rb')].each{ |f| require f }
+#require_relative '../spec/sign_up_helper'
```
Comment (SirNicholas) [inline spec/spec_helper.rb]: please, delete this commented line of code

File: spec/spec_helper.rb [hunk 12]
```diff
@@ -12,6 +14,7 @@
   config.include Capybara::RSpecMatchers
   config.include DataGenerator
 
+
```
Comment (SirNicholas) [inline spec/spec_helper.rb]: Don't commit empty lines

File: spec/support/sign_up_helper.rb [hunk 13]
```diff
@@ -0,0 +1,30 @@
...
+        open.
+        fill_form(email: user.email, password: user.password).
+        submit_form
+    expect(HomePage).to be_authenticated
+    HomePage.wait_for_opened
+  end
+
+end
+
+RSpec.configure do |config|
```
Comment (SirNicholas) [inline spec/support/sign_up_helper.rb]: @romikoops Is this the correct solution here? I am not sure about this. What do you think?

Comment (romikoops) [inline spec/support/sign_up_helper.rb]: everything looks good, except method names. Should start from verbs

**Pull request merged**